# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/ipc/CanvasManagerChild.h
# Commit: 31935f164ef0
# Full Hash: 31935f164ef0d6c4668f211482c139847dcab7f2
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-12-13 00:43:26
# Regressor Bug: 1829026
# File Overlap Count: 1
# Description:
#   Bug 1829026 - Update CanvasTranslator to work with DrawTargetWebgl. r=aosmond
#   
#   This adds the necessary infrastructure for CanvasTranslator to allocate DrawTargetWebgl
#   instead of just allocating TextureData, and to use RemoteTextureMap to handle sending
#   the DrawTargetWebgl frames to the compositor.
# ==============================================================================

diff -r 13c9bf69085f -r 31935f164ef0 gfx/ipc/CanvasManagerChild.h
--- a/gfx/ipc/CanvasManagerChild.h	Tue Dec 12 07:35:02 2023 +0000
+++ b/gfx/ipc/CanvasManagerChild.h	Tue Dec 12 07:35:03 2023 +0000
@@ -49,6 +49,7 @@
   void EndCanvasTransaction();
   void ClearCachedResources();
   void DeactivateCanvas();
+  void BlockCanvas();
 
   RefPtr<layers::CanvasChild> GetCanvasChild();
 
@@ -63,6 +64,7 @@
   RefPtr<webgpu::WebGPUChild> mWebGPUChild;
   const uint32_t mId;
   bool mActive = true;
+  bool mBlocked = false;
 
   static MOZ_THREAD_LOCAL(CanvasManagerChild*) sLocalManager;
   static Atomic<uint32_t> sNextId;